File: service/src/main/java/com/sisweb/api/entity/dto/UsuarioDTO.java
package com.sisweb.api.entity.dto;

import com.fasterxml.jackson.annotation.JsonIgnore;
import com.sisweb.api.entity.Endereco;
import com.sisweb.api.entity.UsuarioPerfil;
import lombok.Getter;
import lombok.Setter;

import java.util.HashSet;
import java.util.Set;

@Getter
@Setter
public class UsuarioDTO {

    private Long id;
    private String login;
    @JsonIgnore
    private String senha;
    private String nome;
    private Boolean ativo;
    private String loginEmailGoogle;
    private String loginEmailGitHub;
    private String loginGitHub;

    private Set<UsuarioPerfil> perfis = new HashSet<>();
    private Endereco endereco;
    Set<String> fone = new HashSet<>();
}

File: cliente/src/models/UsuarioDTO.ts
import { Endereco } from "./Endereco";

export class UsuarioDTO{
    id: number;
    login: String;
    senha: String;
    nome: String;
    ativo: boolean;
    idsPerfis: [];
    fone: [];
    endereco: Endereco;

    setId(id: number){
        this.id = id;
    }

    setLogin(login: String){
        this.login = login;
    }

    setSenha(senha: String){
        this.senha = senha;
    }

    setNome(nome: String){
        this.nome = nome;
    }

    setAtivo(ativo: boolean){
        this.ativo = ativo;
    }

    setPerfis(idsPerfis: []){
        this.idsPerfis = idsPerfis;
    }

    setFone(fone: []){
        this.fone = fone;
    }

    setEndereco(endereco: Endereco){
        this.endereco = endereco;
    }
}